app/src/lib/stores/helpers/branch-pruner.ts

```diff
@@ -0,0 +1,142 @@
+import { Repository } from '../../../models/repository'
+import { RepositoriesStore } from '../repositories-store'
+import { IRepositoryStateCache } from '../repository-state-cache'
+import { Branch, BranchType } from '../../../models/branch'
+import { GitStoreCache } from '../git-store-cache'
+import { getMergedBranches } from '../../git'
+import { fatalError } from '../../fatal-error'
+import { enableBranchPruning } from '../../feature-flag'
+
+/** Check if a repo needs to be pruned at least every 4 hours */
+const BackgroundPruneMinimumInterval = 1000 * 60 * 60 * 4
+
+export class BranchPruner {
+  private timer: NodeJS.Timer | null = null
```

Comment:
This should be `number | null` and we should use `window.setTimeout`. That's what's actually gonna get stored into that variable regardless of what type information we set on it but there's a conflict of types.